core/src/main/java/io/questdb/cairo/wal/WalPurgeJob.java

```diff
@@ -63,6 +65,28 @@ public class WalPurgeJob extends SynchronizedJob implements Closeable {
     private long last = 0;
     private TableToken tableToken;
 
+    private static class SegmentKey {
+        static int encode(int segmentId, boolean pendingTasks, boolean locked) {
```

Comment:
nit: if you do encoding/decoding, why not encode/decode from longs we store instead of ints leaving encoding/decoding of 2 ints into 1 long inline?

Reply:
Currently `sequencerWalIDSegmentIDPairs` uses a different encoding format since it does not need to carry `locked` or `pendingTasks` information.

I.e., conceptually speaking (if we used types that cause GC) the types would be:

```java
class WalIDSegmentIdPair { int walId; int segmentId; }
private final List<WalIDSegmentIdPair> sequencerWalIDSegmentIDPairs = new ArrayList<>();

class WalIdSegmentInfoPair { int walId; int segmentId; boolean pendingTasks; boolean locked; }
private final List<WalIdSegmentInfoPair> onDiskWalIDSegmentIDPairs = new ArrayList<>();
```

Shoehorning both to carry the same fields is artificial and pretty confusing when reading the code.